feature-libs/quote/components/items/quote-items.component.ts

```diff
@@ -5,50 +5,84 @@
 import { UserIdService } from '@spartacus/core';
-import { Quote, QuoteFacade } from '@spartacus/quote/root';
+import { QuoteFacade } from '@spartacus/quote/root';
 import { ICON_TYPE } from '@spartacus/storefront';
-import { NEVER, Observable, Subscription, of, zip } from 'rxjs';
-import { switchMap, tap } from 'rxjs/operators';
+import { Observable, Subscription, combineLatest, of, zip } from 'rxjs';
+import { filter, map, switchMap, tap } from 'rxjs/operators';
 import { QuoteItemsComponentService } from './quote-items.component.service';
 
+interface QuoteItemsData {
+  entries: OrderEntry[] | undefined;
+  readOnly: boolean;
+}
+
 @Component({
   selector: 'cx-quote-items',
   templateUrl: './quote-items.component.html',
+  providers: [
+    AbstractOrderContextSource,
+    { provide: AbstractOrderContext, useExisting: AbstractOrderContextSource },
+  ],
 })
 export class QuoteItemsComponent {
   protected quoteFacade = inject(QuoteFacade);
   protected activeCartFacade = inject(ActiveCartFacade);
   protected multiCartFacade = inject(MultiCartFacade);
   protected quoteItemsComponentService = inject(QuoteItemsComponentService);
   protected userIdService = inject(UserIdService);
+  protected abstractOrderContextSource = inject(AbstractOrderContextSource);
 
-  quoteDetails$: Observable<Quote> = this.quoteFacade.getQuoteDetails();
-  quoteCartReadOnly$: Observable<Cart> = this.userIdService.takeUserId().pipe(
-    switchMap((userId) => zip(this.quoteDetails$, of(userId))),
-    tap(([quote, userId]) => {
-      if (quote.cartId !== undefined) {
-        this.multiCartFacade.loadCart({
-          userId,
-          cartId: quote.cartId,
-          extraData: { active: false },
-        });
-      }
-    }),
-    switchMap(([quote, _userId]) => {
-      return quote.cartId !== undefined
-        ? this.multiCartFacade.getCart(quote.cartId)
-        : NEVER;
-    })
-  );
+  quoteItemsData$: Observable<QuoteItemsData> = this.userIdService
+    .takeUserId()
+    .pipe(
+      switchMap((userId) =>
+        combineLatest([this.quoteFacade.getQuoteDetails(), of(userId)])
+      ),
+      tap(([quote, userId]) => {
+        if (quote.cartId && !quote.isEditable) {
+          this.multiCartFacade.loadCart({
+            userId,
+            cartId: quote.cartId,
+            extraData: { active: false },
+          });
+        }
+      }),
+      switchMap(([quote, _userId]) => {
+        if (!quote.cartId) {
+          this.abstractOrderContextSource.id$.next(quote.code);
+          this.abstractOrderContextSource.type$.next(AbstractOrderType.QUOTE);
+          return zip(of(quote), of(true));
+        } else if (!quote.isEditable) {
+          this.abstractOrderContextSource.id$.next(quote.cartId);
+          this.abstractOrderContextSource.type$.next(
+            AbstractOrderType.SAVED_CART
+          );
+          return combineLatest([
+            this.multiCartFacade.getCart(quote.cartId),
+            of(true),
```

Comment:
nitpick: for readability instead of `true` I suggest passing an object with semantic key, e.g. `{readOnly: true}`. Otherwise the reader has to deduce the meaning of this `true` only from the code in the following operators - where this boolean is named _readonly and readonly

Reply:
Yes, agree